include/boost/url/decode_opts.hpp

```diff
@@ -0,0 +1,44 @@
+//
+// Copyright (c) 2019 Vinnie Falco (vinnie.falco@gmail.com)
+// Copyright (c) 2022 Alan Freitas (alandefreitas@gmail.com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/CPPAlliance/url
+//
+
+#ifndef BOOST_URL_DECODE_OPTS_HPP
+#define BOOST_URL_DECODE_OPTS_HPP
+
+namespace boost {
+namespace urls {
+
+/** Options for removing percent-encoding from strings
+
+    @see
+        @ref decode
+*/
+struct decode_opts
```

Comment:
Why did we split pct_encoding.hpp up into 4 files? I'm not saying not to do it, but I'm wondering what the rationale is.

Reply:
decode_opts had to go into its own file, then the rest followed from there.